test/steps/kogitoapp.go

```diff
@@ -68,77 +69,90 @@ func (data *Data) deployQuarkusExampleServiceWithNativeAndLabels(contextDir, nat
+	return framework.DeployService(data.Namespace, framework.GetDefaultInstallerType(),
 		framework.KogitoAppDeployment{
-			AppName:     filepath.Base(contextDir),
-			ContextDir:  contextDir,
-			Runtime:     v1alpha1.QuarkusRuntimeType,
-			Native:      framework.MustParseEnabledDisabled(native),
-			Persistence: true,
-			Events:      false,
-			Labels:      nil,
+			AppName:            filepath.Base(contextDir),
+			GitSourceURI:       config.GetExamplesRepositoryURI(),
+			GitSourceReference: config.GetExamplesRepositoryRef(),
+			ContextDir:         contextDir,
+			Runtime:            v1alpha1.QuarkusRuntimeType,
+			Native:             framework.MustParseEnabledDisabled(native),
+			Persistence:        true,
+			Events:             false,
+			Labels:             nil,
 		})
 }
 
 func (data *Data) deployQuarkusExampleServiceWithNativeAndPersistenceAndEvents(contextDir, native string) error {
-	return framework.DeployExample(data.Namespace, framework.GetDefaultInstallerType(),
+	return framework.DeployService(data.Namespace, framework.GetDefaultInstallerType(),
 		framework.KogitoAppDeployment{
-			AppName:     filepath.Base(contextDir),
-			ContextDir:  contextDir,
-			Runtime:     v1alpha1.QuarkusRuntimeType,
-			Native:      framework.MustParseEnabledDisabled(native),
-			Persistence: true,
-			Events:      true,
-			Labels:      nil,
+			AppName:            filepath.Base(contextDir),
+			GitSourceURI:       config.GetExamplesRepositoryURI(),
+			GitSourceReference: config.GetExamplesRepositoryRef(),
+			ContextDir:         contextDir,
+			Runtime:            v1alpha1.QuarkusRuntimeType,
+			Native:             framework.MustParseEnabledDisabled(native),
+			Persistence:        true,
+			Events:             true,
+			Labels:             nil,
 		})
 }
 
 func (data *Data) deploySpringBootExampleService(contextDir string) error {
-	return framework.DeployExample(data.Namespace, framework.GetDefaultInstallerType(),
+	return framework.DeployService(data.Namespace, framework.GetDefaultInstallerType(),
 		framework.KogitoAppDeployment{
-			AppName:     filepath.Base(contextDir),
-			ContextDir:  contextDir,
-			Runtime:     v1alpha1.SpringbootRuntimeType,
-			Native:      false,
-			Persistence: false,
-			Events:      false,
-			Labels:      nil,
+			AppName:            filepath.Base(contextDir),
+			GitSourceURI:       config.GetExamplesRepositoryURI(),
+			GitSourceReference: config.GetExamplesRepositoryRef(),
+			ContextDir:         contextDir,
+			Runtime:            v1alpha1.SpringbootRuntimeType,
+			Native:             false,
+			Persistence:        false,
+			Events:             false,
+			Labels:             nil,
 		})
 }
 
 func (data *Data) deploySpringBootExampleServiceWithPersistence(contextDir string) error {
-	return framework.DeployExample(data.Namespace, framework.GetDefaultInstallerType(),
+	return framework.DeployService(data.Namespace, framework.GetDefaultInstallerType(),
 		framework.KogitoAppDeployment{
```

Comment:
You guys recreated the `KogitoApp` structure in the framework? Why?

Reply:
That was mostly prepared by Tristan, but AFAIK it serves as a value holder for both CR and CLI deployments.

Reply:
Understood. Well, I think you could reconsider using the `KogitoApp` instead since it's basically the same thing. Could be a hell to maintain and to keep up with the API. On the other hand, I understand that a wrapper could help in such scenarios.

Reply:
Hmm, I think it could have sense.
@radtriste What do you think about it?

Reply:
indeed ... we could use the KogitoApp struct directly.
I was trying to figure out if there would be anything extra we need, but we can still give as a parameter :)